src/test/java/fi/fmi/avi/archiver/message/populator/ProductMessageTypesValidatorTest.java

```diff
@@ -1,91 +1,135 @@
-        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(TEST_METADATA).buildPartial(), builder);
+        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(METAR_METADATA).buildPartial(), builder);
         assertThat(builder.getProcessingResult()).isEqualTo(ProcessingResult.OK);
     }
 
     @Test
     void another_product_identifier_with_taf() {
         final ArchiveAviationMessage.Builder builder = ArchiveAviationMessage.builder().setType(MessagePopulatorTests.TypeId.TAF.getId());
-        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(ANOTHER_METADATA).buildPartial(), builder);
+        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(OTHER_METADATA).buildPartial(), builder);
         assertThat(builder.getProcessingResult()).isEqualTo(ProcessingResult.OK);
     }
 
     @Test
     void invalid_taf() {
         final ArchiveAviationMessage.Builder builder = ArchiveAviationMessage.builder().setType(MessagePopulatorTests.TypeId.TAF.getId());
-        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(TEST_METADATA).buildPartial(), builder);
+        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(METAR_METADATA).buildPartial(), builder);
         assertThat(builder.getProcessingResult()).isEqualTo(ProcessingResult.FORBIDDEN_MESSAGE_TYPE);
     }
 
+    @Test
+    void valid_taf() {
+        final ArchiveAviationMessage.Builder builder = ArchiveAviationMessage.builder().setType(MessagePopulatorTests.TypeId.TAF.getId());
+        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(TAF_METADATA).buildPartial(), builder);
+        assertThat(builder.getProcessingResult()).isEqualTo(ProcessingResult.OK);
+    }
+
     @Test
     void invalid_type() {
         final ArchiveAviationMessage.Builder builder = ArchiveAviationMessage.builder().setType(-1);
-        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(TEST_METADATA).buildPartial(), builder);
+        productMessageTypesValidator.populate(InputAviationMessage.builder().setFileMetadata(METAR_METADATA).buildPartial(), builder);
         assertThat(builder.getProcessingResult()).isEqualTo(ProcessingResult.FORBIDDEN_MESSAGE_TYPE);
     }
 
+    @Test
+    void misconfigured_message_type() {
+        assertThrows(IllegalArgumentException.class, () ->
+                new ProductMessageTypesValidator(MessagePopulatorTests.TYPE_IDS, PRODUCTS, ImmutableMap.of(TAF_PRODUCT, Collections.singleton(new MessageType("test")))));
+
+    }
+
+    @Test
+    void misconfigured_product() {
+        assertThrows(IllegalArgumentException.class, () ->
+                new ProductMessageTypesValidator(MessagePopulatorTests.TYPE_IDS, PRODUCTS, ImmutableMap.of(OTHER_PRODUCT, Collections.singleton(MessageType.METAR))));
```

Comment:
For consistency, I would suggest using AssertJ assertion.

```suggestion
        assertThatIllegalArgumentException().isThrownBy(() -> new ProductMessageTypesValidator(MessagePopulatorTests.TYPE_IDS, PRODUCTS,
                ImmutableMap.of(OTHER_PRODUCT, Collections.singleton(MessageType.METAR))));
```